From 4557779660924843a3645694e5295a71a07a7fe2 Mon Sep 17 00:00:00 2001
From: Calvin Wan <calvinwan@google.com>
Date: Tue, 6 Jun 2023 19:48:37 +0000
Subject: [PATCH 1568/2871] strbuf: clarify API boundary

strbuf, as a generic and widely used structure across the codebase,
should be limited as a library to only interact with primitives. Add
documentation so future functions can appropriately be placed. Older
functions that do not follow this boundary should eventually be moved or
refactored.

Signed-off-by: Calvin Wan <calvinwan@google.com>
Signed-off-by: Junio C Hamano <gitster@pobox.com>
---
 strbuf.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/strbuf.h b/strbuf.h
index 3dfeadb44c..70778c6e10 100644
--- a/strbuf.h
+++ b/strbuf.h
@@ -1,6 +1,14 @@
 #ifndef STRBUF_H
 #define STRBUF_H
 
+/*
+ * NOTE FOR STRBUF DEVELOPERS
+ *
+ * strbuf is a low-level primitive; as such it should interact only
+ * with other low-level primitives. Do not introduce new functions
+ * which interact with higher-level APIs.
+ */
+
 struct string_list;
 
 /**
-- 
2.39.3 (Apple Git-145)

